src/main/js/src/components/Quarantine/Form.js

```diff
@@ -0,0 +1,214 @@
+    first_name: Yup.string().typeError(t('error.string')).required(t('error.mandatory')),
+    last_name: Yup.string().typeError(t('error.string')).required(t('error.mandatory')),
+    diary_number: Yup.string().typeError(t('error.string')).required(t('error.mandatory')),
+  });
+
+  const isEndDateValid = !R.isNil(endDate) && !R.isEmpty(endDate);
+  const isBirthdateValid = !R.isNil(birthdate) && !R.isEmpty(birthdate);
+  const onFormSubmit = (values) => {
+    // Datepicker uses different date format
+    const parsedBirthdate = birthdate
+          ? dateToString(moment(birthdate, DATE_FORMAT))
+          : null;
+    const parsedEndDate = endDate
+          ? dateToString(moment(endDate, DATE_FORMAT))
+          : null;
+    const payload = {
+      ...values,
+      email: valueOrNull(values.email),
+      phone_number: valueOrNull(values.phone_number),
+      first_name: valueOrNull(values.first_name),
+      last_name: valueOrNull(values.last_name),
+      diary_number: valueOrNull(values.diary_number),
+      birthdate: parsedBirthdate,
+      end_date: parsedEndDate,
+    };
+
+    values.id
+      ? onEdit(payload)
+      : onAdd(payload);
+  };
+
+  const errorMsg = <span className={classes.ErrorMessage}>{t('error.mandatory')}</span>;
+
+  return (
+    <Formik
+      key={`quarantine-form-${form.id}`}
+      initialValues={form}
+      validationSchema={validationSchema}
+      onSubmit={onFormSubmit}
+      render={({ values, handleChange, isValid, dirty }) => (
+        <Form>
+          <div className={classes.QuarantineFormFields}>
+            <div className={classes.QuarantineFormField}>
+              <label htmlFor="language_code">
+                {t('common.examLanguage')}
+              </label>
+              <select
+                value={values.language_code}
+                name="language_code"
+                onChange={handleChange}
+                id="language_code"
+                tabIndex="1">
+                {LANGUAGES.map((lang) => (
+                  <option key={`lang-option-${lang.code}`} value={lang.code}>
+                    {lang.name}
+                  </option>
+                ))}
+              </select>
+            </div>
+
+            <div className={classes.QuarantineFormField}>
+              <label htmlFor="end_date">{t('common.expires')}</label>
+              <DatePicker
+                options={{
+                  defaultDate: endDate,
+                  value: endDate,
+                  minDate: today,
+                  allowInput: true,
+                  dateFormat: DATE_FORMAT_PICKER,
+                  noMinDateUpdate: true,
+                }}
+                autoComplete="off"
+                tabIndex="2"
+                id="end_date"
+                locale={i18n.language}
+                onChange={(dates) => setEndDate(moment(dates[0]).format(DATE_FORMAT))}
+              />
+              {!isEndDateValid && errorMsg}
+            </div>
```

Comment:
Nää datepickerit toimii hieman hassusti nyt (vai edelleen?). Esim. jos tekstin pakottaa manuaalisesti tyhjäksi niin ilmeisesti tilaan jää kuitenkin edellinen validi päivämäärä. Tällöin myöskään validaatiovirhettä ei näytetä ja päivityksessä palvelimelle lähetettävä sisältö ei täsmää näkyvillä olevaan.